src/drivers/simulink/simulink.py

```diff
@@ -0,0 +1,172 @@
+            self._connection.workspace[self._rto_name] = self._runtime_object
+
+            return True
+
+        except Exception as e:
+            print(e)
+            self.sendDebugInfo('SETUP failed: ' + str(e))
+
+        return False
+
+
+    def disconnect(self):
+        """ Disconnect driver. Remove the temporary rto variable. """
+        self._connection.eval('clear ' + self._rto_name, nargout=0)
+        if self.autorun_simulink:
+            self._connection.eval("set_param('simulink_test1','SimulationCommand','Stop');", nargout=0)
+        self._connection.exit()
+
+
+    def addVariables(self, variables: dict):
+        """ Add variables to the driver. Correctly added variables will be added to internal dictionary 'variables'.
+        Any error adding a variable should be communicated to the server using sendDebugInfo() method.
+        : param variables: Variables to add in a dict following the setup format. (See documentation)
+
+        """
+        self.variables.update(variables)
+        for var_id in self.variables:
+            self.variables[var_id]['value'] = self.defaultVariableValue(self.variables[var_id]['datatype'], self.variables[var_id]['size'])
+
+
+    def readVariables(self, variables: list) -> list:
+        """ Read given variable values. In case that the read is not possible or generates an error BAD quality should be returned.
+        : param variables: List of variable ids to be read.
+        : returns: list of tupples including (var_id, var_value, VariableQuality)
+        """
+        res = []
+        for var_id in variables:
+            
+            try:
+                assert self._runtime_object, "Simulation not running"
+                new_value = self._connection.eval(self._rto_name + f'.OutputPort({var_id}).Data')
+                if self.variables[var_id]['size'] > 1:
+                    new_value = [self.getValueFromString(self.variables[var_id]['datatype'], i) for i in new_value[0]]
+                else:
+                    new_value = self.getValueFromString(self.variables[var_id]['datatype'], new_value)
+                res.append((var_id, new_value, VariableQuality.GOOD))
+            except Exception as e:
+                res.append((var_id, self.variables[var_id]['value'], VariableQuality.BAD))
+                self.sendDebugInfo('READ failed: ' + str(e))
+
+
+        return res
+
+
+    def writeVariables(self, variables: list) -> list:
+        """ Write given variable values. In case that the write is not possible or generates an error BAD quality should be returned.
+        : param variables: List of tupples with variable ids and the values to be written (var_id, var_value).
+        : returns: list of tupples including (var_id, var_value, VariableQuality)
+        """
+        res = []
+        for (var_id, new_value) in variables:
+            try:
+                self._connection.eval(f'{self._rto_name}.OutputPort({var_id}).Data={new_value}', nargout=0)
```

Comment:
This is supposed to be `.InputPort`, but that still gives the error `Property 'Data' can only be set for a MATLAB S-function block`, even if I actually use a S-Function block..

Reply:
Fixed this. And turns out it's actually possible to write to S-Function blocks